src/Storages/MergeTree/checkDataPart.cpp

```diff
@@ -163,7 +165,17 @@ static IMergeTreeDataPart::Checksums checkDataPart(
         {
             get_serialization(column)->enumerateStreams([&](const ISerialization::SubstreamPath & substream_path)
             {
-                String file_name = ISerialization::getFileNameForStream(column, substream_path) + ".bin";
+                auto stream_name = ISerialization::getFileNameForStream(column, substream_path);
+                auto file_name = stream_name + ".bin";
+
+                if (!data_part_storage.exists(file_name))
+                    file_name = sipHash128String(stream_name) + ".bin";
```

Comment:
Why do we need this? The file name must be exactly the same as written in the checksums file

Reply:
Here we compute checksums from actual data on disk. If file names diverge from checksums in memory the check will fail as expected.